---
layout: default
title:  'Part 4: Coverage plotting, chimera detection and inspection'
---

# Part 4: Coverage plotting, chimera detection and inspection

As explained during the lectures, the genome amplification process (MDA) results in a coverage bias across the genome and induces chimera formation. 
Here you will have a look at the 'best' assembly that you managed to obtain, and try to assess the level of bias and chimera formation.
In order to visualize the coverage of a normal single-cell assembly, you need to map your reads used in the assembly, back to your assembled contigs. You can then visualize the coverage in Artemis. We will also use a tool to inspect potential chimeras and check the insert size using the pairs. 

4.1 [Reads mapping](scg_part4_1)  
4.2 [Assessing coverage bias](scg_part4_2)  
4.3 [Detection and inspection of chimeric reads](scg_part4_3)  
4.4 [Insert size] (scg_part4_4)  
 
Here are the questions that these results should help you answer:

### Questions
---

**Q4.1** What was the highest coverage for any of your contigs?  
**Q4.2** Do you think that the completeness of your SC genome will improve with more sequence data? Why? Will your genome ever be ‘complete’?  
**Q4.3** How would you explain the coverage patterns you can see on the contigs called NODE_7 and NODE_4.  
**Q4.4** How many chimeric reads did you find in your original dataset? Do you think this value is a true representation of the total amount of chimeras?  
**Q4.5** What was the average insert size in this dataset? 

<div>
 <span style="float:left"><a class="btn btn-primary" href="scg_part3_7"> Previous page</a></span>
 <span style="float:right"><a class="btn btn-primary" href="scg_part4_1"> Next page</a></span>
</div> 
