---
layout: default
title:  'Part 2: Familiarizing yourself with the data'
---

# Part 2: Familiarizing yourself with the data

It is important to be able to recognize what sequence data looks like and how to convert them to different formats to be able to utilize various bioinformatics tools for data analysis. 
The files provided for your analysis are in _Fastq format_, and these are so-called _raw reads_&nbsp;. 
Look into the directory where the data files are located.  
For example, dataset 1.

```sh
ls /proj/g2015028/nobackup/single_cell_exercises/sequences/dataset1/
```

Take a look at what a _Fastq_ file looks like.

```sh
less /proj/g2015028/nobackup/single_cell_exercises/sequences/dataset1/G5_Hiseq_1.fastq
```

**"less"** command allows you to take a look at a text file without changing its contents. 
You can use arrow keys to scroll up and down the window. Type ```q``` to exit. 
Now, we will do a few exercises to see what we have in these raw reads. 
<!--- Illumina produces paired-end reads that always come in pairs and have the same number of reads in each pair. -->
In our practice, there are 2 data sets reads generated by HiSeq 2000 and MiSeq sequencing instruments. 

* HiSeq 2000 read pairs are named as *G5_Hiseq_1.fastq* and *G5_Hiseq_2.fastq*  
* MiSeq read pairs are named as *G5_Miseq_1.fastq* and *G5_Miseq_2.fastq*.  

For example, to count the reads from the HiSeq data set, type:

```sh
grep -c -e "^@HWI" /proj/g2015028/nobackup/single_cell_exercises/sequences/dataset1/G5_Hiseq_1.fastq
```

```grep``` : searches for lines that begin with a string of characters that you are looking for. In this example, the string of characters are __@HWI__.  
```-c``` flag counts the lines containing this string of characters.  
```-e``` flag means you are using Regular Expressions to search for the pattern.  
In this case, ```^``` character means you expect to only check for lines that __begin__ with these specified characters.  
Count how many reads there are in the first Fastq file. Do the same for the second Fastq file (with the *_2.fastq*). 
Write down in a seperate file the number of reads for each file.  

<!--- Can you notice any differences in the read pair file?  -->
Type the following commands to see the 10 first reads ID :

```sh
grep -e "^@HWI" /proj/g2015028/nobackup/single_cell_exercises/sequences/dataset1/G5_Hiseq_1.fastq | head
grep -e "^@HWI" /proj/g2015028/nobackup/single_cell_exercises/sequences/dataset1/G5_Hiseq_2.fastq | head
```

Try to identify the similarities and differences between the *_1.fastq* and *_2.fastq* files  
Next, to count the number of sequences in MiSeq data, type:

```sh
grep -c -e "^@MISEQ" /proj/g2015028/nobackup/single_cell_exercises/sequences/dataset2/G5_Miseq_1.fastq
```

MiSeq instrument produces different Fastq headers from HiSeq instrument and the headers start with _@MISEQ_ . 
Do the same for the second MiSeq Fastq file. Write the total number of reads for the MiSeq data set in the table. 
Add the count of each read pair to get the total number.

### Questions:
---

__Q2.1:__ Looking at a fastq file, can you tell a repeating pattern in the file?  
Do you notice any unique characters or strings that would help you identify where a read begins and ends?  

__Q2.2:__ Did you notice anything similar/different in the Fastq headers of the two read pairs?   
How can you identify paired-end reads?  



<!---

__Q2.3:__ (BONUS) Did you see where the actual DNA sequences are? Can you count how long the DNA sequences are?  
A quick example command to count the length of a DNA sequence:

```sh
echo "ACGTACGT" | awk '{print length($1)}'
head -n4  /proj/g2015028/nobackup/single_cell_exercises/sequences/dataset2/G5_Miseq_1.fastq | sed -n 2p | awk '{print length($1)}'
```

-->