---
layout: default
title:  'Metagenomics Workshop - November 2015'
---
 

# Introduction to Metagenomics and Single Cell analysis

Held in Uppsala November 24-25, 2015.  

## Important Links

[Pre-course material](precourse)  
[Schedule](schedule)  

## Maps & Travel Info

[Map over Uppsala](https://www.google.se/maps/place/Uppsala/@59.8332794,17.6584471,12z/data=!3m1!4b1!4m2!3m1!1s0x465fcbfb8532ab8d:0xaa4fe90a85820807)  
[Map over Uppsala Biomedical Centre (BMC)](http://www.bmc.uu.se/digitalAssets/205/205659_3bmc-2014-810x374.jpg)  
[Travel Info](travel)  
[BMC - Angstrom](https://www.google.com/maps/dir/Uppsala+biomedicinska+centrum,+Uppsala,+Sverige/Caf%C3%A9+%C3%85ngstr%C3%B6m,+Regementsv%C3%A4gen,+Uppsala,+Sweden/@59.840417,17.64003,16z/data=!4m19!4m18!1m10!1m1!1s0x465fcbe795170dad:0x966fefc5e2c0736a!2m2!1d17.634659!2d59.84219!3m4!1m2!1d17.643132!2d59.842019!3s0x465fcbe0e3ce1081:0xbc512f845c408830!1m5!1m1!1s0x465fcbe0014e3e31:0x44c024a73860ee06!2m2!1d17.6470787!2d59.8389771!3e2?hl=sv)  
[TO DO AND EAT in Uppsala](http://www.destinationuppsala.se/en/)  

## Helpful Things

[Linux cheat sheet](../common/images/linux-cheat-sheet.pdf)  
[UPPMAX cheat sheet](../common/images/uppmax-cheat-sheet.png)  
[How to create a webexport in your project](https://www.uppmax.uu.se/webexport-guide)  

## External Resources

[Mac keyboard layouts](../common/mac-keyboard)  
[IGV (Integrative Genomics Viewer) Home](https://www.broadinstitute.org/igv/)  
[Samtools Home](http://www.htslib.org/)  
[BWA Home](http://bio-bwa.sourceforge.net/)  
[GATK Home](https://www.broadinstitute.org/gatk/)  
[SeqAnswers - Next-generation sequencing forum](http://seqanswers.com/)  
[Biosupport.se - Sweden centered bioinformatics Q&A](https://biosupport.se/)   
[Trinity](http://trinityrnaseq.github.io/)  
[Cufflinks](http://cole-trapnell-lab.github.io/cufflinks/)  
[Allpaths-lg](http://www.broadinstitute.org/software/allpaths-lg/blog/?page_id=12)  
[BLAT](http://genome.ucsc.edu/goldenPath/help/blatSpec.html)  
[BLAST](http://www.ncbi.nlm.nih.gov/books/NBK279690/)  
A good explanation of the FASTQ format can be found over at the [MAQ web page](http://maq.sourceforge.net/fastq.shtml). 
It also includes simple Perl code to encode and decode the quality scores. 
